### PR TITLE
 Added Sign in and Register Pages

### PR DESCRIPTION
**Pull request by Lz-01**
Solved the issue #21  . Please Review the Code and give me the corresponding Feedback. Please also check the Github Actions, there are some problems with the test and build.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,4 @@
 import './App.css';
-import {
-  NotificationContainer,
-} from 'react-notifications';
```
Comment (wandergithub) [inline src/App.js:4]: - [ ] Please undo these changes to the notifications. We need them for the pop-up functionality.

File: src/App.js [hunk 2]
```diff
@@ -25,8 +25,10 @@ function App() {
         <Route path="e-commerce/" element={<Home />} />
         <Route path="/items" element={<Items />} />
         <Route path="/customer_service" element={<CustomerService />} />
+        <Route path="/register" element={<RegisterForm />} />
+        <Route path="/login" element={<LoginForm />} />
+        <Route path="/profile" element={<Profile />} />
       </Routes>
-      <NotificationContainer />
```
Comment (wandergithub) [inline src/App.js:29]: - [x] Please undo these changes to the notifications. We need them for the pop-up functionality.

File: src/components/DesktopBar.js [hunk 3]
```diff
@@ -9,69 +10,45 @@ import { PropTypes } from 'prop-types';
 import { Link } from 'react-router-dom';
 import Cart from './Cart';
 import logo from '../assets/logos/logo.png';
-import 'react-notifications/lib/notifications.css';
-import createNotification from './shared/Notifications';
```
Comment (wandergithub) [inline src/components/DesktopBar.js]: - [ ] Please undo these changes to the notifications. We need them for the pop-up functionality.

File: src/components/DesktopBar.js [hunk 4]
```diff
@@ -9,69 +10,45 @@ import { PropTypes } from 'prop-types';
...
-            disabled
-          >
-            <SearchIcon color="primary." />
+          <IconButton type="button" sx={{ p: '10px' }} aria-label="search">
+            <SearchIcon color="primary" />
           </IconButton>
         </Paper>
       </div>
       <div className="language">
-        <IconButton
```
Comment (wandergithub) [inline src/components/DesktopBar.js]: - [ ] Please keep the scope of your changes just to the sign-in/log-in page.

File: src/components/LoginForm.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+              Password
+              label="Password"
+            />
+          </div>
+          <div className="center"><Button variant="outlined">Login</Button></div>
+          <Link to="/register" className="link">
+            <div className="center"><Button variant="outlined">Register</Button></div>
+          </Link>
+
+          <div className="center"><Button variant="outlined">Forgot Password</Button></div>
```
Comment (wandergithub) [inline src/components/LoginForm.js]: These suggestions apply to the Sign-Up form and Login:
- [ ] Please add some margins to the elements so they have a space between them.
- [ ] Add some margin to the div containing the form.
- [ ] Arrange all the items in a column.

Here you can see an example of a layout:
image

File: src/components/NavBar.js [hunk 6]
```diff
@@ -15,19 +14,19 @@ const NavBar = () => {
...
-              <Button variant="outlined">
+              <Button variant="text">Today&apos;s Deals</Button>
+              <Button variant="text">
                 <Link to="customer_service/" className="link">
                   Customer Service
                 </Link>
               </Button>
-              <Button variant="outlined" onClick={createNotification('info')} disabled>Registry</Button>
-              <Button variant="outlined" onClick={createNotification('info')} disabled>Gift cards</Button>
-              <Button variant="outlined" onClick={createNotification('info')} disabled>Sell</Button>
```
Comment (wandergithub) [inline src/components/NavBar.js:30]: - [x] Kindly, don't enable all the buttons, just change the Registry one: `<Button variant="text">Registry</Button>`  `<Button variant="outlined" onClick={createNotification('info')} disabled>Registry</Button>`

**Review by wandergithub — CHANGES_REQUESTED** [review]
Hi @Lz-01 Great job so far, I suggest some changes before merging this PR. Here is the overall explanation:

- [ ] So, is very important to keep the buttons disabled, so users know this functionality is still in production.
- [ ] Also keep all your changes restricted to form registration.
- [ ] Please restore the react-notification that was previously implemented.
- [ ] I also commented on some suggestions about the layout, they are just minor enhancement changes.

Check comments for more details.

If you have any questions or comments, remember to tag me so I can see them.

File: src/App.js (hunk 2 above)
Comment (Lz-01) [reply src/App.js:29]: Done! But i get the error message :"warning: LF will be replaced by CRLF in src/App.js.
The file will have its original line endings in your working directory"
ESLint: 'createNotification' is defined but never used.(no-unused-vars)

File: src/App.js (hunk 1 above)
Comment (Lz-01) [reply src/App.js:4]: Done! But i get the error message :"warning: LF will be replaced by CRLF in src/App.js.
The file will have its original line endings in your working directory"
ESLint: 'createNotification' is defined but never used.(no-unused-vars)

File: src/components/DesktopBar.js (hunk 4 above)
Comment (Lz-01) [reply src/components/DesktopBar.js]: Done!

File: src/components/LoginForm.js (hunk 5 above)
Comment (Lz-01) [reply src/components/LoginForm.js]: Done!

File: src/components/NavBar.js (hunk 6 above)
Comment (Lz-01) [reply src/components/NavBar.js:30]: Done!

File: src/components/RegisterForm.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+          border: '1px dashed grey',
+        }}
+        noValidate
+        autoComplete="off"
+      >
+        <Typography textAlign="center" variant="h4" gutterBottom>
+          Sign Up
+        </Typography>
+        <Typography textAlign="center" variant="subtitle1" gutterBottom>
+          It's free and only takes a minute
```
Comment (wandergithub) [inline src/components/RegisterForm.js:23]: So, this is an ESlinter error which you can solve by escaping the `'` sign like this:

`It&apos;s free and only takes a minute`

That is because of these kinds of characters and `&` for example. Can be harmless code, meaning that this can be used to harm the page look on the HTML code. And Also, to prevent XSS attacks.

File: src/App.js (hunk 2 above)
Comment (wandergithub) [reply src/App.js:29]: You can solve the line ending with: `npx eslint . --fix`

**Review by wandergithub — APPROVED** [review]
## ✔️ Great Work @Lz-01 

In the comments, I left some suggestions that I think would be good to keep in mind in your future coding experience.

I also encourage you to take a look at the commits I made before merging your PR, so you can see how I implemented these small changes.

Happy coding and thank you for your work 🤝 .
